app/models/user.rb

```diff
@@ -224,6 +224,26 @@ def archived?
     archived_email.present?
   end
 
+  def self.dedup_users!
+    select_users_to_archive.each { |u| u.archive! }
+  end
+
+  def self.select_users_to_archive
+    ids = left_joins(:participant_identities).where(participant_identities: { user_id: nil }).pluck(:id)
+
+    ids_to_exclude = []
+    ids_to_exclude << AdminProfile.where(user_id: ids).pluck(:user_id)
+    ids_to_exclude << AppropriateBodyProfile.where(user_id: ids).pluck(:user_id)
+    ids_to_exclude << DeliveryPartnerProfile.where(user_id: ids).pluck(:user_id)
+    ids_to_exclude << FinanceProfile.where(user_id: ids).pluck(:user_id)
+    ids_to_exclude << InductionCoordinatorProfile.where(user_id: ids).pluck(:user_id)
+    ids_to_exclude << LeadProviderProfile.where(user_id: ids).pluck(:user_id)
+
+    ids = ids - ids_to_exclude.flatten.uniq
+    User.where(id: ids)
+  end
+
+
```

Comment:
Could we move this to a service class instead?  Would feel more natural to me to invoke this via a service but what do you think?